modules/programs/starship.nix

```diff
@@ -0,0 +1,91 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.starship;
+
+  configFile = config:
+    pkgs.runCommand "config.toml"
+      {
+        buildInputs = [ pkgs.remarshal ];
+        preferLocalBuild = true;
+        allowSubstitutes = false;
+      }
+      ''
+        remarshal -if json -of toml \
+          < ${pkgs.writeText "config.json" (builtins.toJSON config)} \
+          > $out
+      '';
+in
+
+{
+  meta.maintainers = [ maintainers.marsam ];
+
+  options.programs.starship = {
+    enable = mkEnableOption "starship";
+
+    settings = mkOption {
+      type = types.attrs;
+      default = {};
+      description = ''
+        Configuration written to
+        <filename>~/.config/starship.toml</filename>.
+        </para><para>
+        See <link xlink:href="https://starship.rs/config/" /> for the full list
+        of options.
+      '';
+    };
+
+    enableBashIntegration = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to enable Bash integration.
+      '';
+    };
+
+    enableZshIntegration = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to enable Zsh integration.
+      '';
+    };
+
+    enableFishIntegration = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to enable Fish integration.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.starship ];
+
+    xdg.configFile."starship.toml" = mkIf (cfg.settings != {}) {
+      source = configFile cfg.settings;
+    };
+
+    programs.bash.initExtra = mkIf cfg.enableBashIntegration ''
+      if [ -z "$INSIDE_EMACS" ]; then
```

Comment:
the INSIDE_EMACS looks arbitrary :s why disable it there ? the name should be more generic IMO

Reply:
It's to avoid problems with Emacs tramp, nixpkgs also has the same check https://github.com/NixOS/nixpkgs/blob/076860e0340a5e4a909b9a710e186508b14d1c90/nixos/modules/programs/bash/bash.nix#L98

Reply:
interesting. Sounds like a hack to work around a bug in emacs. emacs should ignore unknown control characters. Have you tried without the check ? 
Anyway I guess it's fine having it then.

Reply:
`$INSIDE_EMACS` is not defined for me when I use tramp. Could it also test that `"$TERM" != "dumb"` ?